lib/rust/ensogl/core/src/debug/stats.rs

```diff
@@ -15,48 +15,137 @@
 
 use enso_prelude::*;
 
+use enso_web::Performance;
 use js_sys::ArrayBuffer;
 use js_sys::WebAssembly::Memory;
 use wasm_bindgen::JsCast;
 
 
 
+// ====================
+// === TimeProvider ===
+// ====================
+
+/// An entity that can retrieve current time.
+pub trait TimeProvider {
+    /// Returns current time, measured in milliseconds.
+    fn now(&self) -> f64;
+}
+
+impl TimeProvider for Performance {
+    fn now(&self) -> f64 {
+        self.now()
+    }
+}
```

Comment:
This should be refactored to other abstractions over web-functions (I dont remember the module where we keep them).